### DISCUSSION TITLE
How can I open my app from a URI scheme (Windows)

### DISCUSSION THREAD

**Discussion by balaedit** [airsdk/Adobe-Runtime-Support#2845] [General]
Is there anyway to open my air application via chrome and also needs to know how to register my custom uri in windows.

**Comment by balaedit** [comment]
@ajwfrost @marchbold  pls help here for windows.

**Comment by marchbold** [comment]
@balaedit This is probably one for your installer as I don't think there's anything in AIR you need to specifically do here, you should be able to get details from the invoke event, though I haven't personally tried this. 

To register your app to a custom url scheme you need to read the following:

https://learn.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/platform-apis/aa767914(v=vs.85)?redirectedfrom=MSDN

You can also setup "apps for websites" to open from an http(s) address. 

https://learn.microsoft.com/en-us/windows/uwp/launch-resume/web-to-app-linking

**Reply by balaedit** [reply]
I had checked this links while research but Is there any way to integrate in our AIR app descriptor files so that it will automatically register while installing the app as we describe in mobile app descriptor file.

**Reply by marchbold** [reply]
Not that I'm aware of, really depends how you create your windows installer etc I would have thought.

**Reply by balaedit** [reply]
I am not creating any windows Installer instead creating the air Installer package. But needs to know how to add uri scheme in app descriptor file for windows air package as we add in air Android apps descriptor file.

**Reply by ajwfrost** [reply]
Are you creating a .air file then? or using ADT to create a native installer package (target `native`)?

We don't have a way to set this up automatically on Windows, like there is on Android/iOS. The best option might be to create this as a bundle (target `bundle`) and then create a Windows installation application e.g. using WiX, and as part of that you can get it to register a custom URI scheme to open up your app - e.g. see https://stackoverflow.com/questions/8304851/wix-how-to-register-an-application-to-a-url-protocol

We can look at this as a feature request though, I think it would be a fairly handy feature!

thanks

**Comment by pnico** [comment]
If you are doing a UWP app for Windows Store using Desktop Bridge (or I guess now MSIX?) you can update the UWP app manifest
https://learn.microsoft.com/en-us/windows/uwp/launch-resume/handle-uri-activation and then you don't need to update the registry in your installer which I think will probably pop up the dialog asking to do something with admin privileges. MSIX might also let you do something similar if it's not a UWP / Windows Store app, I don't know. For just a plain windows EXE I think you'll need to have an installer that will add the keys to the registry (I don't know if the AIR packager for Windows adds an app manifest?)

Note this is different from "[web-to-app-linking](https://learn.microsoft.com/en-us/windows/uwp/launch-resume/web-to-app-linking)" which is a regular https:// url like you can do on android and iOS and requires you to host a config at the domain to prove you own it. On Windows (most browsers anyway) that only works outside the browser (like in a messaging app or if another app navigates to that url in code) so it's not useful if you want to open the app from a web page. The older, less-secure "my-custom-scheme://some/deep/link" method actually does work in the browser though, if the required registry keys or UWP app manifest entries are there. (Edit: it was a long time ago so I don't remember if AIR will also pass the correct deep link to the invoke event)
